### PR TITLE
Update closure style guidelines

### PR DESCRIPTION
**Pull request by mayasaxena**
Update closure style guidelines to match Swiftlint rules

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -582,7 +582,7 @@ let layout = { (view1: UIView, view2: UIView) in
 
 ### Void arguments/return types
 
-It is preferable to not use `Void` in closure arguments or return types, omitting parentheses where possible.
+It is preferable to omit `Void` in closure return types.
```
Comment (Benuuu) [inline README.md]: This line should still include `closure arguments` e.g. in the example `noArgClosure`

**Review by alexpersian — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -593,29 +593,28 @@ let noReturnClosure = { (arg) in doSomething(with: arg) } // void return type, o
...
-let noArgClosure = { (Void) -> Int in return getValue() } // don't do this
-let noReturnClosure = { (arg) -> Void in doSomething(with: arg) } // don't do this
+let noArgNoReturnClosure = { (Void) -> Void in doSomething() }
+let noArgClosure = { (Void) -> Int in return getValue() }
+let noReturnClosure = { (arg) -> Void in doSomething(with: arg) }
 ```
+**_Rationale:_** A `Void` return type can be inferred, thus it is unnecessarily verbose to include it
 
-However, when defining closure type, `Void` is preferred to `()`
+When defining closure type, prefer `()` for parameters, and `Void` for return types.
```
Comment (Benuuu) [inline README.md:610]: Do we have a rationale for keeping `Void` in the return types?

Comment (mayasaxena) [reply README.md:610]: The SwiftLint issue relating to that rule linked to a post saying that it matches Apple APIs, and that `Void` is clearer to read than `()`
http://ericasadun.com/2015/05/11/swift-vs-void/

Comment (Benuuu) [reply README.md:610]: Could you add that rationale?

**Review by Benuuu — APPROVED** [review]

**Comment by tzm41** [comment]
Haha great. Swift 4 will give warnings for the wrong style `(Void)`.

**Pull request merged**